File: java-api/src/main/java/de/davelee/personalman/api/ChangePasswordRequest.java
package de.davelee.personalman.api;

import lombok.*;

/**
 * This class is part of the PersonalMan REST API. It represents a request to change the password of a particular user.
 * @author Dave Lee
 */
@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ChangePasswordRequest {

    // company associated with
    private String company;

    // username who's password should be changed
    private String username;

    // The token of the user to verify that they are logged in
    private String token;

    // current password for this user
    private String currentPassword;

    // new password to set for this user
    private String newPassword;

}

File: server/src/users/requests/changepassword.request.ts
import {ApiProperty} from "@nestjs/swagger";

export class ChangePasswordRequest {

    // company associated with
    @ApiProperty()
    private company: string;

    // username who's password should be changed
    @ApiProperty()
    private username: string;

    // The token of the user to verify that they are logged in
    @ApiProperty()
    private token: string;

    // current password for this user
    @ApiProperty()
    private currentPassword: string;

    // new password to set for this user
    @ApiProperty()
    private newPassword: string;

    getCompany(): string {
        return this.company;
    }

    getUsername(): string {
        return this.username;
    }

    getToken(): string {
        return this.token;
    }

    getCurrentPassword(): string {
        return this.currentPassword;
    }

    getNewPassword(): string {
        return this.newPassword;
    }
}